src/xrGame/ui/UILoadingScreenHardcoded.h

```diff
@@ -164,34 +164,33 @@ R"(<w>
 inline pcstr GetLoadingScreenXML()
 {
     const bool is_wide = UICore::is_widescreen();
-    if (ShadowOfChernobylMode)
+    const pcstr loadingScreenType = pSettingsOpenXRay->read_if_exists<pcstr>("gameplay", "loading_screen", "cop");
+
+    if (loadingScreenType == "soc" || loadingScreenType == "shoc")
     {
         if (is_wide)
             return LoadingScreenXML16x9ShadowOfChernobyl;
         return LoadingScreenXMLClearSky;
-    }
-    if (ClearSkyMode)
+    } else if (loadingScreenType == "cs")
     {
         if (is_wide)
             return LoadingScreenXML16x9ClearSky;
         return LoadingScreenXMLClearSky;
+    } else
+    {
+        if (is_wide)
+            return LoadingScreenXML16x9;
+        return LoadingScreenXML;
     }
-    // Call of Pripyat
-    if (is_wide)
-        return LoadingScreenXML16x9;
-    return LoadingScreenXML;
 }
 
 inline pcstr GetLoadingScreenTexturesDescr()
 {
-    if (ShadowOfChernobylMode)
-    {
-        return LoadingScreenXMLClearSkyTexturesDescription;
-    }
-    if (ClearSkyMode)
-    {
+    const pcstr loadingScreenType = pSettingsOpenXRay->read_if_exists<pcstr>("gameplay", "loading_screen", "cop");
+
+    if (loadingScreenType == "soc" || loadingScreenType == "shoc" || loadingScreenType == "cs")
         return LoadingScreenXMLClearSkyTexturesDescription;
```

Comment:
Correct raw strings comparison should be done using `strcmp` function. Since the code `strcmp(...) == 0` may be harder to read, I returned old verbose `if` checks to make it easier to read the code.
```suggestion
    cpcstr loadingScreenType = pSettingsOpenXRay->read_if_exists<pcstr>("ui", "loading_screen", "cop");

    if (xr_strcmp(loadingScreenType, "soc") == 0 || xr_strcmp(loadingScreenType, "shoc") == 0)
    {
        return LoadingScreenXMLClearSkyTexturesDescription;
    }
    if (xr_strcmp(loadingScreenType, "cs") == 0)
    {
        return LoadingScreenXMLClearSkyTexturesDescription;
    }
```